components/ucd/normal/tests/conformance_tests.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright 2017 The UNIC Project Developers.
+//
+// See the COPYRIGHT file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+
+extern crate unic_ucd_normal;
+extern crate unic_ucd_utils;
+
+use unic_ucd_normal::DecompositionType as DT;
+use unic_ucd_utils::iter_all_chars;
+use std::collections::HashSet as Set;
+use std::{char, u32};
+
+
+const DT_TEST_DATA: &'static str =
+    include_str!("../../../../data/ucd/test/DecompositionTypeTest.txt");
+
+
+#[derive(Debug)]
+struct Fail {
+    pub line_num: Option<usize>,
+    pub char: char,
+    pub exp_dt: Option<DT>,
+    pub actual_dt: Option<DT>,
+}
+
+#[test]
+#[should_panic(expected = "11172 test cases failed! (1100892 passed)")]
+fn test_decomposition_type_conformance() {
+    let mut passed_num: u32 = 0;
+    let mut failures: Vec<Fail> = Vec::new();
+    let mut to_consider: Set<char> = iter_all_chars().collect();
+
+    for (line_idx, line) in DT_TEST_DATA.lines().enumerate() {
+        // Remove comments and trailing whitespace
+        let line = if let Some(idx) = line.find('#') {
+            &line[0..idx]
+        } else {
+            line
+        }.trim();
+
+        // Ignore empty lines
+        if line.is_empty() {
+            continue;
+        }
+
+        // State setting lines
+        if line.starts_with('@') {
+            panic!("State setting line in DecompositionTypeTest.txt; expected none");
+        }
+
+        // Data lines
+        {
+            let fields: Vec<_> = line.split(';').map(str::trim).collect();
```

Comment:
Yeah, probably would get better perf if not using `collect()` and use `nth()` to access. But, since it's just a test and not part of the lib, it's fine either way.